mobile_verifier/src/verified_subscriber_mapping_event.rs

```diff
@@ -0,0 +1,146 @@
+use crate::Settings;
+use chrono::{DateTime, Duration, Utc};
+use file_store::{
+    file_info_poller::{FileInfoStream, LookbackBehavior},
+    file_source,
+    verified_subscriber_mapping_event::VerifiedSubscriberMappingEvent,
+    verified_subscriber_mapping_event_ingest_report::VerifiedSubscriberMappingEventIngestReport,
+    FileStore, FileType,
+};
+use futures::stream::StreamExt;
+use sqlx::{Pool, Postgres, Transaction};
+use std::ops::Range;
+use task_manager::{ManagedTask, TaskManager};
+use tokio::sync::mpsc::Receiver;
+
+pub struct VerifiedSubscriberMappingEventDeamon {
+    pool: Pool<Postgres>,
+    events: Receiver<FileInfoStream<VerifiedSubscriberMappingEventIngestReport>>,
+}
+
+impl VerifiedSubscriberMappingEventDeamon {
+    pub fn new(
+        pool: Pool<Postgres>,
+        events: Receiver<FileInfoStream<VerifiedSubscriberMappingEventIngestReport>>,
+    ) -> Self {
+        Self { pool, events }
+    }
+
+    pub async fn create_managed_task(
+        pool: Pool<Postgres>,
+        settings: &Settings,
+        file_store: FileStore,
+    ) -> anyhow::Result<impl ManagedTask> {
+        let (events, event_server) =
+            file_source::continuous_source::<VerifiedSubscriberMappingEventIngestReport, _>()
+                .state(pool.clone())
+                .store(file_store)
+                .lookback(LookbackBehavior::StartAfter(settings.start_after))
+                .prefix(FileType::VerifiedSubscriberMappingEventIngestReportV1.to_string())
+                .create()
+                .await?;
+
+        let task = Self::new(pool, events);
+
+        Ok(TaskManager::builder()
+            .add_task(event_server)
+            .add_task(task)
+            .build())
+    }
+
+    pub async fn run(mut self, shutdown: triggered::Listener) -> anyhow::Result<()> {
+        tracing::info!("Starting sme deamon");
+        loop {
+            tokio::select! {
+                biased;
+                _ = shutdown.clone() => {
+                    tracing::info!("sme deamon shutting down");
+                    break;
+                }
+                Some(file) = self.events.recv() => {
+                    self.process_file(file).await?;
+                }
+            }
+        }
+        Ok(())
+    }
+
+    async fn process_file(
+        &self,
+        file: FileInfoStream<VerifiedSubscriberMappingEventIngestReport>,
+    ) -> anyhow::Result<()> {
+        tracing::info!(
+            "Processing Verified Mapping Event file {}",
+            file.file_info.key
+        );
+        let mut transaction = self.pool.begin().await?;
+        let mut reports = file.into_stream(&mut transaction).await?;
+
+        while let Some(report) = reports.next().await {
+            save_event(&report.report, &mut transaction).await?;
```

Comment:
We need to verify the `carrier_mapping_key` is a trusted key before saving to database.

Reply:
done